BUILD.md

```diff
@@ -1,44 +1,72 @@
-Build Instructions
-==================
+# Build Instructions
 
 # Contents
 
-* [Note](#note)
-* [Getting the source](#source)
-* [Linux](#linux)
-* [Windows](#windows)
-* [macOS](#macos)
-
+- [Getting the source](#getting-the-source)
+- [Linux](#linux)
+- [Windows](#windows)
+- [macOS](#macos)
 
 # Getting the source
 
 Clone the source code using git and grab all the submodules:
 
 ```
 git clone https://github.com/PolyMC/PolyMC.git
+cd PolyMC
 git submodule init
 git submodule update
 ```
 
+The rest of the documentation assumes you have already cloned the repository.
+
 # Linux
 
 Getting the project to build and run on Linux is easy if you use any modern and up-to-date linux distribution.
 
 ## Build dependencies
-* A C++ compiler capable of building C++11 code.
-* Qt Development tools 5.6 or newer  (`qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools libqt5core5a libqt5network5 libqt5gui5` on Debian-based system)
-* cmake 3.1 or newer (`cmake` on Debian-based system)
-* zlib (`zlib1g-dev` on Debian-based system)
-* Java JDK (`openjdk-17-jdk`on Debian-based system)
-* GL headers (`libgl1-mesa-dev` on Debian-based system)
+
+- A C++ compiler capable of building C++11 code.
+- Qt Development tools 5.6 or newer (`qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools libqt5core5a libqt5network5 libqt5gui5` on Debian-based system)
+- cmake 3.1 or newer (`cmake` on Debian-based system)
+- zlib (`zlib1g-dev` on Debian-based system)
+- Java JDK (`openjdk-17-jdk`on Debian-based system)
+- GL headers (`libgl1-mesa-dev` on Debian-based system)
+
+You can use IDEs like KDevelop or QtCreator to open the CMake project if you want to work on the code.
+
+### Building a portable binary
+
+```sh
+# make all the folders
+mkdir build
+mkdir install
+# configure the project
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=../install ..
+# build
+make -j$(nproc) install
+```
+
+### Building & Installing to the System
+
+This is the preferred method for installation, and is suitable for packages.
+
+```sh
+# configure everything
+cmake -S . -B build \
+	-DCMAKE_BUILD_TYPE=Release \
```

Comment:
nit: The indentation could probably be fixed here while we're changing it anyway